src/copy_tree.rs

```diff
@@ -38,38 +40,71 @@ pub fn copy_tree<ST: ReadTree, DT: WriteTree>(
         // deleted or changed while this is running.
         ui::set_bytes_total(source.size()?.file_bytes);
     }
+
+    // Target selected for copy (subpaths from here and included)
+    let target = &options.restore_only;
+    let target_tree: Vec<&str> = target.split('/').collect();
+
     ui::set_progress_phase("Copying");
     for entry in source.iter_entries()? {
-        if options.print_filenames {
-            crate::ui::println(entry.apath());
-        }
-        ui::set_progress_file(entry.apath());
-        if let Err(e) = match entry.kind() {
-            Kind::Dir => {
-                stats.directories += 1;
-                dest.copy_dir(&entry)
-            }
-            Kind::File => {
-                stats.files += 1;
-                dest.copy_file(&entry, source).map(|s| stats += s)
+        // Check if this entry is selected for copy
+        let subtree: Vec<&str> = entry.apath().split('/').collect();
```

Comment:
I would suggest to do this matching with an `Apath::is_prefix_of` method, which can then be unit tested.